explainer.md

```diff
@@ -431,130 +409,90 @@ should be at the discretion of the user agent. For example, the site shouldn't
-
-* Sites that only want to badge their app icon or only want to badge their tab
-  could just concern themselves with the relevant API.
-* The "document" API could potentially support badges that aren't supported by
-  host operating systems, such as arbitrary Unicode characters.
+A user agent could provide the `setAppBadge` API without `setClientBadge`, which
+means the above check will apply the fallback, while app icons can still be
+badged.
 
 ## Detailed API proposal
 
 ### The model
 
-A badge is associated with a [scope](https://www.w3.org/TR/appmanifest/#navigation-scope).
-
-* Documents are badged with the most specific badge for their URL (i.e. prefer a badge for `/page/1` to a badge for `/page/` when on the url `/page/1?foo=7`).
-* For [installed applications](https://www.w3.org/TR/appmanifest/#installable-web-applications), a user agent **MAY** display the badge with the most specific scope still encompassing the [navigation scope](https://www.w3.org/TR/appmanifest/#navigation-scope) of the application in an [OS specific context](#OS-Specific-Contexts).
+A badge is associated with either a document, or an [installed app](https://www.w3.org/TR/appmanifest/#installable-web-applications).
 
-At any time, the badge for a specific scope, if it is set, may be either:
+At any time, the badge for a specific document or app, if it is set, may be either:
 
 * A "flag" indicating the presence of a badge with no contents, or
 * A positive integer.
 
 The model does not allow a badge to be a negative integer, or the integer value 0 (setting the badge to 0 is equivalent to clearing the badge).
 
-The user agent is allowed to clear the badge whenever there are no foreground pages open on the origin (the intention of this is so that when the user agent quits, it does not need to serialize all the badge data and restore it on start-up; sites should re-apply the badge when they open).
+The user agent is allowed to clear all badges on an origin whenever there are no foreground pages open on the origin (the intention of this is so that when the user agent quits, it does not need to serialize all the badge data and restore it on start-up; sites should re-apply the badge when they open).
 
 ### The API
 
-The `Badge` interface is a member object on
-[`Window`](https://html.spec.whatwg.org/#the-window-object) and
-[`ServiceWorkerGlobalScope`](https://w3c.github.io/ServiceWorker/#serviceworkerglobalscope-interface).
-It contains the following methods:
-
-* `Badge.set([contents], [options])`: Sets the badge for the scope in *options*
-  to *contents* (an integer), or to "flag" if *contents* is omitted. If
-  *contents* is 0, clears the badge for the given scope.
-* `Badge.clear([options])`: Clears the badge for the scope in *options*.
-* `Badge.canBadgeDocument()`: Returns true if the user agent would display a
-  badge applying to a document scope on or near the page's favicon.
+The Badge API methods are members of interface is a member object on
```

Comment:
I think the API definition is redundant now (all of this has already been mentioned above)... plus it's starting to look like a spec. We should just add to the spec. Risk is that this explainer will quickly fall out of date.

Reply:
I think (at least for now) this serves a useful purpose of being in between the usage examples and the (non-existent) spec.

The usage examples show the basic usage but don't describe what the methods do and their arguments.

The spec will be very dense and state algorithms, not an informal description.

I like having an informal description of the methods (which is why I deleted the WebIDL and went back to this "conversational" documentation. Eventually this should be the role of MDN or similar, but for now, I'd like to keep it here.